icevision/models/torchvision_models/faster_rcnn/show_results.py

```diff
@@ -31,3 +36,103 @@ def show_results(
         denormalize_fn=denormalize_fn,
         show=show,
     )
+
+
+def get_losses(
+    model: nn.Module,
+    dataset: Dataset,
+) -> Tuple[List[dict], dict]:
+    """
+    Gets a dataset and a model as input and returns losses calculated per image + losses stats.
+    In case of faster_rcnn those are loss_classifier, loss_box_reg and loss_total
+    (sum of the first 2 plus RPN related losses, e.g. sum of 4 terms).
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+
+    Returns
+    -------
+    samples_plus_losses: those are the samples (dictionaries) contained in the original dataset
+                          enriched with losses.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+    device = torch.device("cpu")
+    model = model.to(device)
+    dl = valid_dl(dataset, batch_size=1, num_workers=0, shuffle=False)
+
+    torch.manual_seed(0)
+    samples_plus_losses = []
+    losses_stats = {
+        "loss_classifier": [],
+        "loss_box_reg": [],
+        "loss_objectness": [],
+        "loss_rpn_box_reg": [],
+        "loss_total": [],
+    }
+    for_text = ["loss_classifier", "loss_box_reg", "loss_total"]
+
+    with torch.no_grad():
+        for (x, y), sample in pbar(dl):
+            loss = model(x, y)
+            loss = {k: float(v.numpy()) for k, v in loss.items()}
+            loss["loss_total"] = sum(loss.values())
+
+            text = ""
```

Comment:
Is it possible somehow to move the text logic to be inside the plot function? Or maybe there is an advantage to keep it here?

Reply:
Need to look into this.

Reply:
Also, `text` (e.g. annotations) is added to the sample dict, to facilitate retrieving the info later on.

Reply:
I see your point, the only thing I have in my head is that `text` is very specific to the plotting functionality, while reading the code it's not so intuitive why this field is being built. When we get to `plot_top_losses` is also not so intuitive to find where the `annotations` are being created from. 

There is nothing wrong in the way you implemented, my only opinion is that having this snippet inside `plot_top_losses` is more intuitive and more immediate to understand what is happening.

Reply:
I understand. Let me see what I can do about this.

Reply:
I split the `text` logic into a separate function and added all the losses to the plot annotations for completeness, given we are providing `total` and/or `weighted` losses and it might be confusing what those contain.
This is the new viz.
![image](https://user-images.githubusercontent.com/1206519/102318411-90ed5e00-3f79-11eb-8019-d4fc82ac60e5.png)

Reply:
Let me know your thoughts.

Reply:
Looks really nice!!! Can you post this image on discord so we see what other people think?

Reply:
Done